### PR TITLE
Support writing lists of bytes in zict.File

### PR DESCRIPTION
**Pull request by mrocklin**
This is to reduce byte copying in dask.distributed

### CODE REVIEW

**Comment by mrocklin** [comment]
cc @pitrou for review (should be fast)

**Comment by pitrou** [comment]
Is it a problem if File is the only class to support such API variation?

File: zict/file.py [hunk 1]
```diff
@@ -68,7 +78,11 @@ def __getitem__(self, key):
 
     def __setitem__(self, key, value):
         with open(os.path.join(self.directory, _safe_key(key)), 'wb') as f:
-            f.write(value)
+            if isinstance(value, (tuple, list)):
+                for v in value:
+                    f.write(v)
```
Comment (pitrou) [inline zict/file.py]: You can probably write `f.writelines(value)` instead, though I don't think that will bring much speedup.

**Comment by mrocklin** [comment]
> Is it a problem if File is the only class to support such API variation?

It would be nicer if zip and others supported this as well.  I don't personally care much though.  I would add them as needed.

**Comment by mrocklin** [comment]
Planning to merge soon.

**Pull request merged**